File: Baekjoon/구현/1764.py
# https://www.acmicpc.net/problem/1764
# 5분정도는 문제가 뭔소린가 했더니 ㅈㄴ 웃기네

import sys
input = sys.stdin.readline

n, m = map(int, input().split())

a = set(input().rstrip() for _ in range(n))
b = set(input().rstrip() for _ in range(m))
    
c = sorted(list(a & b))

print(len(c))
for i in c:
    print(i)

File: Baekjoon/구현/1764.cpp
// https://www.acmicpc.net/problem/1764

#include <iostream>
#include <vector>
#include <algorithm>
#include <map>
using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    map<string, bool> check;
    for (int i = 0; i < n; i++) {
        string name;
        cin >> name;
        check[name] = true;
    }
    vector<string> v;
    for (int i = 0; i < m; i++) {
        string name;
        cin >> name;
        if (check[name]) {
            v.push_back(name);
        }
    }
    sort(v.begin(), v.end());
    cout << v.size() << '\n';
    for (auto &name : v) {
        cout << name << '\n';
    }
    return 0;
}